modules/reference/pages/properties/object-storage-properties.adoc

```diff
@@ -33,6 +33,7 @@ Optional API endpoint.
 
 - AWS: When blank, this is automatically generated using <<cloud_storage_region,region>> and <<cloud_storage_bucket,bucket>>. Otherwise, this uses the value assigned.
 - GCP: Uses `storage.googleapis.com`.
+- Azure: If you have enabled hierarchical namespaces for your storage account and use a custom endpoint, use <<cloud_storage_azure_adls_endpoint,`cloud_storage_azure_adls_endpoint`>>.
```

Comment:
Currently in our [Tiered Storage doc](https://docs.redpanda.com/current/manage/tiered-storage/?tab=tabs-1-microsoft-absadls#configure-object-storage) we have a note that says

For hierarchical namespaces created with a custom endpoint, set cloud_storage_azure_adls_endpoint and cloud_storage_azure_adls_port. If you haven’t configured custom endpoints in Azure, there’s no need to edit these properties.

Should both 1) HNS enabled and 2) custom endpoints set up be True in order to use cloud_storage_azure_adls_endpoint and cloud_storage_azure_adls_port? (Is it possible for users to have HNS enabled, but not have custom endpoints for their Azure storage account?)
If Azure users do not have HNS enabled, do they then need to configure cloud_storage_api_endpoint (as with AWS/GCP users)?
We specify that these endpoint properties are "Optional: No", but at the same time we say "Default: null" (and cloud_storage_api_endpoint is also [described](https://docs.redpanda.com/current/reference/properties/object-storage-properties/#cloud_storage_api_endpoint) as "Optional API endpoint"). This is confusing for readers. Can we clarify what config properties are required to use Azure for tiered storage vs what are optional, and in which scenarios?

Reply:
so it's a bit messy but the jist is that:

redpanda uses an endpoint to connect to Azure Blob Storage (the main one). 
Users can enable HNS in their storage account, and if so, redpanda will auto-discover it and use an additional endpoint to perform List/Delete operations instead of the main one.

these endpoints are in the form `[cloud_storage_azure_storage_account].blob.core.windows.net` for main and `[cloud_storage_azure_storage_account].dfs.core.windows.net` for the additional HNS endpoint.

In addition, users can have custom endpoints (I think, for example, when they deploy in China), and in that case, they can use them by setting `cloud_storage_api_endopoint` for the main and `cloud_storage_azure_adls_endpoint` if they have HNS enabled. 

Note: `cloud_storage_api_endopoint` and `cloud_storage_azure_adls_endpoint` are independent of one another, we don't require that `cloud_storage_api_endopoint` is set if `cloud_storage_azure_adls_endpoint` is set, thought in a real-world scenario if you have HNS enabled and  `cloud_storage_api_endopoint` set, i would expect that `cloud_storage_azure_adls_endpoint` too would need to be set.

If HNS is not enabled, `cloud_storage_azure_adls_endpoint` it's just ignored

> We specify that these endpoint properties are "Optional: No",
this seems wrong, redpanda will fallback to the default `[cloud_storage_azure_storage_account].blob.core.windows.net` [...] without problems